src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.MultiAgent.cs

```diff
@@ -1229,7 +1229,6 @@ private static ulong TransferDataFromRequestStream(IntPtr buffer, int length, Ea
                     asyncRead = easy._requestContentStream.ReadAsync(
                        new Memory<byte>(sts.Buffer, 0, Math.Min(sts.Buffer.Length, length)), easy._cancellationToken);
                 }
-                Debug.Assert(asyncRead != null, "Badly implemented stream returned a null task from ReadAsync");
```

Comment:
Why was this deleted?

Reply:
`ValueTask<T>` is a struct, so it's impossible for a method to return a null one. However, if `default(ValueTask<int>)` cannot be awaited (or is otherwise not valid), it seems like an assertion could say but have a different form.

Reply:
So, I didn't see the type here. It's fine to remove the assert. This is likely left over from when it was a Task.